cli/command.go

```diff
@@ -274,19 +274,46 @@ func (c *BaseCommand) Hidden() bool {
 
 // Prompt prompts the user for a value. If stdin is a tty, it prompts. Otherwise
 // it reads from the reader.
-func (c *BaseCommand) Prompt(msg string) (string, error) {
-	scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+func (c *BaseCommand) Prompt(ctx context.Context, msg string, args ...any) (string, error) {
```

Comment:
If canceled, this function leaks a goroutine that is reading from stdin. That could be a problem if there are any further reads from stdin. Later reads will behave weirdly since the read they're trying to do might get stolen by this orphaned goroutine.

Suggestions:
 - document this behavior and leave it as-is. Say something like "if the context is canceled, this function leaves stdin in a bad state."
 - Interrupt the background goroutine reading from stdin by closing stdin (or some other way of using Close()) to interrupt the background goroutine. Not sure if this works.
 - Somehow implement a cancelable io.Reader in pkg

Reply:
Added defer close on os.Stdin

Reply:
Changed to comment on the issue below